main.tf

```diff
@@ -51,7 +51,7 @@ module "exec_label" {
 
 resource "aws_ecs_task_definition" "default" {
   count                    = local.create_task_definition ? 1 : 0
-  family                   = module.this.name
+  family                   = var.ecs_service_name
```

Comment:
`ecs_service_name` isn't a required variable, and can be `null`. 
If you want this the same as the service name (probably a good idea), make sure you're setting it to the same things as the service name. In this case, that would be `var.ecs_service_name != null ? var.ecs_service_name : module.this.id`

Though looking at it again, I think that should actually be `module.service_label.id` (both here and in the actual `aws_ecs_service` resource definition further down. `service_label` sets label_order to `["name", "attributes"]`, so two service task modules with the same "name" but different attributes (ie "web" vs "worker") should have things named uniquely.